### DISCUSSION TITLE
kOps 1.23.2 version

### DISCUSSION THREAD

**Discussion by argoyle** [eddycharly/terraform-provider-kops#582] [General]
Hi! Any plans for a kOps 1.23.2 version? There is an IAM-fix for cluster autoscaler which would be nice for K8S 1.23+.

**Comment by eddycharly** [comment]
Hi, I'll give a look tonight.
Thanks for reporting.

**Reply by argoyle** [reply]
No rush, seemed to be a 1.23.3 version coming quite soon as well so might be fine to wait for that if you are busy elsewhere. 😊

**Reply by eddycharly** [reply]
It's just a matter of merging a couple of PRs and tagging the branch...

**Reply by eddycharly** [reply]
Release has started, it should be available soon :crossed_fingers:

**Reply by argoyle** [reply]
Nice! Just tested and cluster came back up fine. Had to do a cloud_only true though since Terraform claimed I had no changes to apply. Would it be possible to make sure it always does an update if provider version is changed? Perhaps store the last used provider-version in the state or something similar?

**Reply by eddycharly** [reply]
You can always add a dummy key to the keepers but I agree it’s hacky.

**Reply by argoyle** [reply]
I'll see if I can find a nicer way and create a PR. Thanks for creating the provider, it has made it really easy to spin up whole environments for us. ❤️

**Reply by eddycharly** [reply]
If you can get the provider version, I would definitely recommend injecting the info in the `keepers` map.
The `updater` is the most tricky part, it does not map to a real resource in the kOps sense.
Thanks for the kind words, of course PRs are very welcome !

**Reply by argoyle** [reply]
Since you use goreleaser it's possible to get the version information into variables in the code like this in main:
```
var (
	version = "dev"
	commit  = "none"
	date    = "unknown"
)
``` 

Goreleaser sets default ldflags to `-s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}} -X main.builtBy=goreleaser`. 

Then it's just a matter of getting it into the cluster_updater somehow I guess? Not sure how to do that though since I have no prior experience with Terraform providers.

**Reply by argoyle** [reply]
PR created 😄 https://github.com/eddycharly/terraform-provider-kops/pull/601

**Reply by eddycharly** [reply]
`v1.23.5-alpha.3` is being built, with the version changes :crossed_fingers:

**Reply by argoyle** [reply]
🎉 will try it out once build is completed.

**Reply by argoyle** [reply]
Worked perfectly! Thanks again! ❤️
